nbclient/client.py

```diff
@@ -512,7 +534,7 @@ async def async_execute(self, reset_kc=False, **kwargs):
             info_msg = await self.async_wait_for_reply(msg_id)
             self.nb.metadata['language_info'] = info_msg['content']['language_info']
             self.set_widgets_metadata()
-
+        self._update_state(ExecutionState.COMPLETE)
```

Comment:
Should we add `self._update_state(ExecutionState.IDLE)` just after that?

Reply:
I guess idle and complete have some overlap, but they're still separate useful states, right?

Reply:
Yes, `IDLE` always follows `COMPLETE`, except for the first state.

Reply:
makes sense.